### ISSUE TITLE
SPIR-V: Issue with modf

### ISSUE THREAD

**Issue by CraziestBug** [Microsoft/DirectXShaderCompiler#1741]
We're using DXC to generate SPIR-V from HLSL and I've come across a bug with one of our shaders. I was able to pin down the issue to the **modf** intrinsic function and sub-component selection.
I've simplified the shader with the following lines:
```
float4 Test(float2 coord : TEXCOORD0) : SV_Target0
{
#if 1
	int3 icoord = 0; // Error
#else
	int2 icoord = 0; // Compile
#endif
	float2 d = saturate( modf(coord, icoord.xy) );
	return float4(icoord.y, 0, 0, 1);
}
```

**Comment by ehsannas** [comment]
Thanks for reporting. Will investigate.

**Comment by ehsannas** [comment]
Acknowledged. I have a fix, but can only apply after PR #1792 is landed.
